client/sites-dashboard/controller.tsx

```diff
@@ -47,53 +47,21 @@ export function sanitizeQueryParameters( context: PageJSContext, next: () => voi
 }
 
 export function maybeSitesDashboard( context: PageJSContext, next: () => void ) {
-	const siteCount = getCurrentUser( context.store.getState() )?.site_count ?? 0;
+	const state = context.store.getState();
+	const siteCount = getCurrentUser( state )?.site_count ?? 0;
 
 	if ( context.query[ 'hosting-flow' ] || siteCount === 0 ) {
-		context.store.dispatch( hideMasterbar() );
-		return emptySites( context, siteCount, next );
+		const newSiteUrl = getAddNewSiteUrl( {
+			isHostingFlow: context.query[ 'hosting-flow' ] === 'true',
+			isDevAccount: getUserSetting( state, 'is_dev_account' ) === true,
+		} );
+
+		return window.location.assign( newSiteUrl );
```

Comment:
This creates a full-page refresh. Don't like it. Using `page.redirect` or `page.replace` doesn't work for some reason: I get a blank page with no errors in the console. 🤔